test/handlers/admin/data-contributions.js

```diff
@@ -83,10 +83,17 @@ describe('admin data contributions handler', () => {
     it('updates the DataContribution and redirects to /admin/data-contributions', () => {
       const payload = {
         approved: true,
+        trial_id: '11111111-1111-1111-1111-111111111111',
+        data_category_id: null,
         curation_comments: 'My comments',
       };
       let dataContributionId;
 
+      factory.create('dataCategory')
+        .then((dataCategory) => {
+          payload.data_category_id = dataCategory.id;
+        });
```

Comment:
There's still the same race condition. This promise needs to be part of the chain that's returned, like:

```javascript
return factory.create('dataContribution')
  .then(() => ())
  .then(() => factory.create('dataCategory'))
  .then((dataCategory) => (payload.data_category_id = dataCategory.attributes.id))
  // ...
```

It's also possible to refactor this using the `Promise.join` like:

```javascript
return Promise.join(
    factory.create('dataContribution'),
    factory.create('dataCategory'),
    (dataContribution, dataCategory) => {
      dataContributionId = dataContribution.attributes.id;
      payload.data_category_id = dataCategory.attributes.id;
    }
  )
  .then(() => server.inject({}))
```